pdc/apps/package/views.py

```diff
@@ -532,3 +534,238 @@ def bulk_update(self, *args, **kwargs):
         of objects to be modified and their values use the same format as normal patch.
         """
         return bulk_operations.bulk_update_impl(self, *args, **kwargs)
+
+
+class ReleasedFilesViewSet(PDCModelViewSet):
+    """
+    API endpoint released files.
+    """
+    queryset = models.ReleasedFiles.objects.all().order_by('id')
+    serializer_class = serializers.ReleasedFilesSerializer
+    filter_class = filters.ReleasedFilesFilter
+    permission_classes = (APIPermission,)
+    docstring_macros = PUT_OPTIONAL_PARAM_WARNING
+
+    def _check_parameters(self, data):
+        pass
+
+    def list(self, *args, **kwargs):
+        """
+        __Method__: GET
+
+        __URL__: $LINK:releasedfiles-list$
+
+        __Query params__:
+
+        %(FILTERS)s
+
+        __Response__: a paged list of following objects
+
+            {
+                "build (optional, default=null)": "string",
+                "created_at (read-only)": "datetime",
+                "file (optional, default=null)": "string",
+                "file_primary_key": "int",
+                "id (read-only)": "int",
+                "obsolete (optional, default=false)": "boolean",
+                "package (optional, default=null)": "string",
+                "release_date": "date",
+                "released_date (optional, default=null)": "date",
+                "repo": {
+                            "arch": "string",
+                            "content_category": "ContentCategory.name",
+                            "content_format": "ContentFormat.name",
+                            "id (read-only)": "int",
+                            "name": "string",
+                            "product_id (optional, default=null, nullable)": "int",
+                            "release_id": "string",
+                            "repo_family": "RepoFamily.name",
+                            "service": "Service.name",
+                            "shadow (optional, default=false)": "boolean",
+                            "variant_uid": "string"
+                        },
+                "updated_at (read-only)": "datetime",
+                "zero_day_release (optional, default=false)": "boolean"
+            }
```

Comment:
Can you use `%(SERIALIZER)s` instead?

Reply:
Try this before, it can't print the repo's data. Just a repo id.

Reply:
Again, see how this is done for [multi-destinations](https://github.com/product-definition-center/product-definition-center/blob/a205c467510145206a4e0853412f1b91c8647fa0/pdc/apps/repository/serializers.py#L147).

Reply:
I think I know what you mean, but as I mentioned I used "%(SERIALIZER)s" before.
But it just can show the response like the data part:
{
    "file_primary_key": "int",
    "release_date": "date",
    "repo": "Repo.id",
    "package (optional, default=null)": "string",
    "build (optional, default=null)": "string",
    "file (optional, default=null)": "string",
    "created_at (optional, default=null)": "date",
    "released_date (optional, default=null)": "date",
    "updated_at (optional, default=null)": "date",
    "obsolete (optional, default=false)": "boolean",
    "zero_day_release (optional, default=false)": "boolean"
}
This is not what I want, I want to show the repo's detail, so write like this.
I saw your "rest_api/v1/multi-destinations/" codes, same like what I wrote before.
Here the response likes below:
{
    "active (optional, default=true)": "boolean", 
    "destination_repo": "Repo", 
    "global_component": "GlobalComponent.name", 
    "id (read-only)": "int", 
    "origin_repo": "Repo", 
    "subscribers (optional, default=[])": [
        "Person.username"
    ]
}
But I think the Repo part should show all the keys, something like release-components.

Reply:
If I misunderstand your mean, please let me know.

Reply:
Well, the issue is that after adding a new field in Repo you'll need to update docstrings everywhere. Please don't make me do it, I'll forget to update it anyway. :)

If it's really needed we could change the code which generates the documentation.

Reply:
Yes, you are right, that's a problem, but I think we need to add such repo fields.
I wrote like this because there are other apis with nested fields.
How about to create another task to track this? Then we can do some researching and maybe update all the related codes then?

Reply:
> How about to create another task to track this? Then we can do some researching and maybe update all the related codes then?

I think, that would take too much time. How about just use `RepoField` type for the "repo" serializer field? :)

Reply:
In my view, I still think it is better to keep it here. 
Because the repo has many columns and I add a pdc_endpoint here and also now the build/file/package are also not in get_fields, so keep it here in response is more clear.
But I have changed the data part to  %(WRITABLE_SERIALIZER)s :)

Reply:
I add the patch of #503 here for good looking and test. Will merge these two together when pass the review.